Source/Scene/GroundPolylineBatch.js

```diff
@@ -0,0 +1,501 @@
+        Pass) {
+    'use strict';
+
+    /**
+     * Renders a batch of polylines that have been subdivided to be draped on terrain.
+     *
+     * @alias GroundPolylineBatch
+     * @constructor
+     * @private
+     *
+     * @param {Object} options An object with following properties:
+     * @param {Float32Array|Uint16Array} options.positions The positions of the polylines
+     * @param {Number[]} options.counts The number or positions in the each polyline.
+     * @param {Number[]} options.widths The width of each polyline.
+     * @param {Cartesian3} [options.center=Cartesian3.ZERO] The RTC center.
+     * @param {Number} [options.quantizedOffset] The quantized offset. If undefined, the positions should be in Float32Array and are not quantized.
+     * @param {Number} [options.quantizedScale] The quantized scale. If undefined, the positions should be in Float32Array and are not quantized.
+     * @param {Cesium3DTileBatchTable} options.batchTable The batch table for the tile containing the batched polylines.
+     * @param {Number[]} options.batchIds The batch ids for each polyline.
+     * @param {BoundingSphere} options.boundingVolume The bounding volume for the entire batch of polylines.
+     */
+    function GroundPolylineBatch(options) {
+        // these arrays are all released after the first update.
+        this._positions = options.positions;
+        this._widths = options.widths;
+        this._counts = options.counts;
+        this._batchIds = options.batchIds;
+
+        this._center = options.center;
+        this._quantizedOffset = options.quantizedOffset;
+        this._quantizedScale = options.quantizedScale;
+
+        this._boundingVolume = options.boundingVolume;
+        this._batchTable = options.batchTable;
+
+        this._va = undefined;
+        this._sp = undefined;
+        this._rs = undefined;
+        this._uniformMap = undefined;
+        this._command = undefined;
+
+        this._spPick = undefined;
+        this._rsPick = undefined;
+        this._pickCommand = undefined;
+
+        this._constantColor = Color.clone(Color.WHITE);
+        this._highlightColor = this._constantColor;
+    }
+
+    var attributeLocations = {
+        previousPosition : 0,
+        currentPosition : 1,
+        nextPosition : 2,
+        expandAndWidth : 3,
+        a_batchId : 4
+    };
+
+    function decodePosition(positions, index, decodeMatrix, center, result) {
+        var encodedPosition = Cartesian3.unpack(positions, index, result);
+        var rtcPosition = Matrix4.multiplyByPoint(decodeMatrix, encodedPosition, encodedPosition);
+        return Cartesian3.add(rtcPosition, center, rtcPosition);
+    }
+
+    var scratchDecodeMatrix = new Matrix4();
+    var scratchP0 = new Cartesian3();
+    var scratchP1 = new Cartesian3();
+    var scratchPrev = new Cartesian3();
+    var scratchCur = new Cartesian3();
+    var scratchNext = new Cartesian3();
+
+    function createVertexArray(primitive, context) {
```

Comment:
I saw your tasklist about moving this to a web worker, which I agree with.  How hot is this in the profile?  Just determining if we should do this now or later.